nn/transformer.py

```diff
@@ -0,0 +1,230 @@
+"""
+Transformer Modules
+"""
+
+from __future__ import annotations
+from typing import Optional, Any, Union, Callable
+from .. import nn
+from . import LayerRef
```

Comment:
Why do you have `LayerRef` as an explicit import? Why not just use `nn.LayerRef`? I think that would be more consistent.

Reply:
I was thinking, that in longer function heads this might become more messy and for typing I felt like this was fine to do, since like this we do have a clear separation between typing and actually using the class. e.g. When `LayerRef` would be used in code it would be via `nn.LayerRef` in the way I am thinking right now.

Reply:
But the difference between `LayerRef` and `nn.LayerRef` is really minor. Also, it should be consistent. Esp it should not be different for typing than for other code.